src/storageProviders/fs/fsStorageProviderBase.ts

```diff
@@ -1,12 +1,22 @@
 import { promises } from 'node:fs';
 import { Logger } from '@map-colonies/js-logger';
+import { inject } from 'tsyringe';
 import { IStorageProvider } from '../iStorageProvider';
 import { IDataLocation } from '../../common/interfaces';
+import { SERVICES } from '../../common/constants';
+import { IConfig } from '../../common/interfaces';
 
 export abstract class FsStorageProviderBase implements IStorageProvider {
-  public constructor(protected batchSize: number, protected logger: Logger) {}
+  private readonly disableSourcesCleanup: boolean;
+  public constructor(@inject(SERVICES.CONFIG) protected readonly config: IConfig, protected batchSize: number, protected logger: Logger) {
+    this.disableSourcesCleanup = this.config.get<boolean>('disableCleanup.sources');
+  }
 
   public async deleteDiscretes(discreteLocationArray: IDataLocation[]): Promise<void> {
+    if (this.disableSourcesCleanup) {
```

Comment:
check if this will prevent tiles deletion when FS is the provider (add test to make sure)

Reply:
i moved it outside from from FsStorageProviderBase to FsSourcesStorageProvider, so it will not prevent tiles deletion.